### DISCUSSION TITLE
Are shake128-len and shake256-len available for CMS signing and verifying?

### DISCUSSION THREAD

**Discussion by mholschbach** [open-quantum-safe/oqs-provider#271] [Q&A]
Hi,

I'm using the openquantumsafe/curl docker image together with the documentation from https://github.com/open-quantum-safe/openssl#cms-demo for the CMS demo which worked very well. I was also able to switch to the SHAKE digest algorithm by using the additional parameter "-md shake256".

But how can I modify the output length of the SHAKE algoritms? 

My search led me to a wide variety of results:
* The output of _openssl list -digest-algorithms_ only shows the currently used OIDs for SHAKE128 and SHAKE256 but not the ones for SHAKE128-LEN and SHAKE256-LEN.
* This comment seems to confirm: "Below two are incomplete OIDs, to be uncommented when we figure out how to handle them..." from https://github.com/openssl/openssl/blob/master/crypto/objects/objects.txt#L1089C1-L1092C52
* But the output of _echo -n "" | openssl dgst -shake256 -xoflen 64_ produces the expected digest value.

Any hints are welcome.

**Comment by baentsch** [comment]
> But how can I modify the output length of the SHAKE algoritms?

I'm afraid I have no idea :-( May it be an idea to ask this to the [OpenSSL community](https://github.com/openssl/openssl/discussions/categories/q-a?discussions_q=is%3Aopen+category%3AQ%26A)?

**Reply by mholschbach** [reply]
Thank you.
For anyone interested, this is the [discussion](https://github.com/openssl/openssl/discussions/22283) at OpenSSL.
